server/src/main/java/org/elasticsearch/index/mapper/IndexFieldMapper.java

```diff
@@ -53,7 +54,12 @@ public String typeName() {
         }
 
         @Override
-        protected boolean matches(String pattern, QueryShardContext context) {
+        protected boolean matches(String pattern, boolean caseInsensitive, QueryShardContext context) {
+            if (caseInsensitive) {
+                // Thankfully, all index names are lower-cased so we don't have to pass a case_insensitive mode flag
+                // down to all the index name-matching logic. We just lower-case the search string
+                pattern = pattern.toLowerCase(Locale.ROOT);
```

Comment:
Should it use a specific locale to avoid lowercasing more than ASCII chars ?

Reply:
Do you know which one would make sense?